src/components/favorite-button/favorite-button.vue

```diff
@@ -0,0 +1,52 @@
+<template>
+  <v-tooltip left>
+    <template v-slot:activator="{ on }">
+      <div @click="handleSetFavorite" v-on="on">
+        <v-icon v-if="favorite" color="yellow">mdi-star</v-icon>
+        <v-icon v-else color="grey lighten-1">mdi-star-outline</v-icon>
+      </div>
+    </template>
+    <span>{{tooltipText}}</span>
+  </v-tooltip>
+</template>
+
+<script>
+export default {
+  name: 'FavoriteButton',
+
+  props: {
+    trackId: {
```

Comment:
This component could work with other favorites scope, not only for tracks, so I suggest removing track prefix, just **id**